digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/Digiroad2Api.scala

```diff
@@ -1439,15 +1526,22 @@ class Digiroad2Api(val roadLinkService: RoadLinkService,
             val roadLink = groupedRoadLinks(asset.linkId).head
             (roadLink, Measures(asset.startMeasure, asset.endMeasure), roadLink.administrativeClass)
           }
-      assetInfo.foreach { case (roadLink, measures, administrativeClass) =>
-        if (!user.isAuthorizedToWriteInArea(maintenanceRoadService.getAssetArea(Some(roadLink), measures), administrativeClass))
-          halt(Unauthorized("User not authorized"))
-      }
+
+        assetInfo.foreach { case (roadLink, measures, administrativeClass) =>
+          if (!user.isAuthorizedToWriteInArea(maintenanceRoadService.getAssetArea(Some(roadLink), measures), administrativeClass))
+            halt(Unauthorized("User not authorized"))
+        }
     } else {
       roadLinks.foreach(a => validateUserAccess(user, Some(typeId))(a.municipalityCode, a.administrativeClass))
     }
   }
 
+  private def validateUserRightsForLanes(linkIds: Set[Long], user: User) : Unit = {
+
+    val roadLinks = roadLinkService.fetchVVHRoadlinksAndComplementary(linkIds)
+    roadLinks.foreach(a => validateUserAccess(user)(a.municipalityCode, a.administrativeClass))
+  }
```

Comment:
When we start using the old approach at serializers we can delete this and start using the many ways already existent to validate the user.

Reply:
Cannot be as requested.
Lanes can come with multiple linkIds not only one as other assets